From b84f786db6d85cbf9973adfccdf135cb71824b87 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Laleve=CC=81e?= <nicolas.lalevee@hibnet.org>
Date: Tue, 28 Oct 2014 00:37:04 +0100
Subject: [PATCH] IVY-1493 : Can't resolve wildcard dependencies when remote
 artifact server does not set content-type header (Thanks to Andrew Bernhagen)

---
 CHANGES.txt                                           | 2 ++
 src/java/org/apache/ivy/util/url/ApacheURLLister.java | 7 ++++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/CHANGES.txt b/CHANGES.txt
index 4d18a9df0..da4a7ffb4 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -22,6 +22,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
 	Andreas Axelsson
 	Stephane Bailliez
 	Karl Baum
+	Andrew Bernhagen
 	Mikkel Bjerg
 	Per Arnold Blaasmo
 	Jeffrey Blattman
@@ -154,6 +155,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
 - FIX: failed to resolve dynamic revisions in some cases for URL repositories (IVY-1472)
 - FIX: ClassCastException in Eclipse 4.4.1 (IVY-1487) (Thanks to Carsten Pfeiffer)
 - FIX: NullPointerException when accessing charset to invalid URL (IVY-1452) (Thanks to Frédéric Riviere)
+- FIX: Can't resolve wildcard dependencies when remote artifact server does not set content-type header (IVY-1493) (Thanks to Andrew Bernhagen)
 
    2.4.0-rc1
 =====================================
diff --git a/src/java/org/apache/ivy/util/url/ApacheURLLister.java b/src/java/org/apache/ivy/util/url/ApacheURLLister.java
index 2df69b084..e2e7dea30 100644
--- a/src/java/org/apache/ivy/util/url/ApacheURLLister.java
+++ b/src/java/org/apache/ivy/util/url/ApacheURLLister.java
@@ -117,7 +117,12 @@ public List retrieveListing(URL url, boolean includeFiles, boolean includeDirect
         String charset = urlInfo.getBodyCharset();
 
         InputStream contentStream = urlHandler.openStream(url);
-        BufferedReader r = new BufferedReader(new InputStreamReader(contentStream, charset));
+        BufferedReader r = null;
+        if (charset == null) {
+            r = new BufferedReader(new InputStreamReader(contentStream));
+        } else {
+            r = new BufferedReader(new InputStreamReader(contentStream, charset));
+        }
 
         String htmlText = FileUtil.readEntirely(r);
 
